testsuite/features/step_definitions/command_steps.rb

```diff
@@ -180,13 +180,18 @@
 end
 
 When(/^I use spacewalk-common-channel to add all "([^"]*)" channels with arch "([^"]*)"$/) do |channel, architecture|
-  channels_to_synchronize = CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, "#{channel}-#{architecture}")
-  raise ScriptError, "Synchronization error, channels for #{channel}-#{architecture} in #{product} not found" if channels_to_synchronize.nil?
+  channels_to_synchronize = CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, channel) ||
+                            CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, "#{channel}-#{architecture}")
+  raise ScriptError, "Synchronization error, channel #{channel} or #{channel}-#{architecture} in #{product} product not found" if channels_to_synchronize.nil? || channels_to_synchronize.empty?
 
   channels_to_synchronize.each do |os_product_version_channel|
-    log "Adding channel: #{os_product_version_channel}"
     command = "spacewalk-common-channels -u admin -p admin -a #{architecture} #{os_product_version_channel.gsub("-#{architecture}", '')}"
     get_target('server').run(command)
+    log "Channel #{os_product_version_channel.gsub("-#{architecture}", '')} added"
+  rescue ScriptError
+    command = "spacewalk-common-channels -u admin -p admin -a #{architecture} #{os_product_version_channel}"
+    get_target('server').run(command)
+    log "Channel #{os_product_version_channel} added"
```

Comment:
If it works, I would not imped to merge this for now.
But personally, I think it's a bad practice to use a rescue section to perform actions that can raise again similar errors.

Let's iterate over in the future if you prefer.

Reply:
The whole problem is inconsistency in channels naming from SCC. The best how I can visualize it is through example (notice `aarch64` specifically):
```
opensuse_leap15_2:   x86_64
opensuse_leap15_2-aarch64: aarch64
```
in 15.5 it's just:
```
opensuse_leap15_5:   x86_64, aarch64
```

I need to handle that.

Reply:
Wouldn't that make the same thing?
```
When(/^I use spacewalk-common-channel to add all "([^"]*)" channels with arch "([^"]*)"$/) do |channel, architecture|
  channels_to_synchronize = CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, channel) ||
                            CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, "#{channel}-#{architecture}")
  
  if channels_to_synchronize.nil? || channels_to_synchronize.empty?
    log "Synchronization error, channel #{channel} or #{channel}-#{architecture} in #{product} product not found" 
    command = "spacewalk-common-channels -u admin -p admin -a #{architecture} #{os_product_version_channel}"
    get_target('server').run(command)
    log "Channel #{os_product_version_channel} added"
  else
    channels_to_synchronize.each do |os_product_version_channel|
      command = "spacewalk-common-channels -u admin -p admin -a #{architecture} #{os_product_version_channel.gsub("-#{architecture}", '')}"
      get_target('server').run(command)
      log "Channel #{os_product_version_channel.gsub("-#{architecture}", '')} added"
    end
  end
end
```

Reply:
In `if` clause you have `os_product_version_channel` undefined, you have to iterate over all channel child elements. Secondly we agreed with @Bischoff to raise an exception instead of just logging [[Ref](https://github.com/uyuni-project/uyuni/pull/8281#discussion_r1483040029)].

Reply:
Some good read about the topic https://rmosolgo.github.io/ruby/programming/2016/11/23/raising-exceptions-is-bad.html

Reply:
As you feel, I don't have the capacity to discuss further this implementation.

Reply:
Under current edited comment I have no problem to add `if`, `else` directives. It may improve clarity, but in my opinion checking the output code is a little bit redundant as I expect only `200` code or `ScriptError` in this particular function. Other errors that may appear are application based, not logic ones. I am open for further discussions, I ping other's if they want to share their opinion on the subject @nodeg @Bischoff

Reply:
I'd implement it somewhat like:

```ruby
When(/^I use spacewalk-common-channel to add all "([^"]*)" channels with arch "([^"]*)"$/) do |os, architecture|
  channels_to_synchronize = CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, os) ||
                            CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, "#{os}-#{architecture}")

  raise ScriptError, "Nothing to synchronize for #{os} or #{os}-#{architecture} in #{product} product." if channels_to_synchronize.nil? || channels_to_synchronize.empty?

  channels_to_synchronize.each do |full_channel_name|
    no_arch_channel_name = full_channel_name.gsub("-#{architecture}", '')

    success = false
    [no_arch_channel_name, full_channel_name].each do |channel_name|
      command = "spacewalk-common-channels -u admin -p admin -a #{architecture} #{channel_name}"
      _out, code = get_target('server').run(command, check_errors: false)
      if code.zero?
        log "Channel #{channel_name} added"
        success = true
        break
    end

    raise ScriptError, "Synchronization error on #{full_channel_name} channel.\nError: #{out}" unless success
  end
end
```

as I'd probably try to avoid the raise-rescuse-raise too

Reply:
Two iterations increase time complexity from O(n) to O(n*m) what is not desirable, but I have managed to find something in between. I need to check it better tomorrow. Let me know what you think of current solution @srbarrios @NamelessOne91